### ISSUE TITLE
Azure Event Hubs client as Azure service or WebApp

### ISSUE THREAD

**Issue by harisbal** [Azure/azure-sdk-for-python#21980]
Could you provide some examples as to how we can run an [event hub receiver/sender](https://github.com/Azure/azure-sdk-for-python/tree/azure-eventhub_5.6.1/sdk/eventhub/azure-eventhub/samples) as an Azure service.
I'm mostly interested in the cases where a continuous connection with the EventHub has to be maintained, therefore Azure Functions do not seem appropriate.

If deploying as a WebApp is the solution, how could that be realised (e.g. using flask or django)?

**Comment by YalinLi0312** [comment]
@harisbal Thanks for your feedback, we'll investigate asap.

**Comment by yunhaoling** [comment]
hey @harisbal , apologize for the late response.

I'm not an expert in django but if I understand it correctly, are you kinda looking seeking for ways to share the event hub client among different modules?

I think there're multiple ways to achieve that:
- let the even thub client be a module level variable

### in eh_client.py

```python
from azure.eventhub import EventHubProducerClient
import os

CONNECTION_STR = os.environ['EVENT_HUB_CONN_STR']
EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']

eh_producer = EventHubProducerClient.from_connection_string(conn_str=CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
```

### in main.py
```python
from eh_client.py import eh_producer  # assuming main.py and eh_client.py are in the same directory, but in reality, the import would be depending on your project hierarchy
eh_producer.send_batch(...)
```


- or let the event hub client be a variable of a class
### in eh_client_wrapper.py

from azure.eventhub import EventHubProducerClient
import os

CONNECTION_STR = os.environ['EVENT_HUB_CONN_STR']
EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']

class WrapperClass:
    eh_producer = EventHubProducerClient.from_connection_string(conn_str=CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
    def __init__(self):
        if not self.eh_producer:
            self.eh_producer = EventHubProducerClient.from_connection_string(conn_str=CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
```

### in main.py
```python
# Initiate the wrapper in the program's entry point
from eh_client_wrapper import WrapperClass # assuming main.py and eh_client.py are in the same directory, but in reality, the 
wrapper_class = WrapperClass()
```

### in some_module.py
from eh_client_wrapper import WrapperClass
WrapperClass.eh_producer.send_batch(...)
```

Let me know whether it answers your questions, and I would be glad to discuss more with you on it.

**Comment by harisbal** [comment]
Thank you @yunhaoling but I'm still not sure if this answers my question. Let me rephrase the problem.
I have an external data source that produces data and streams them through a websocket. For that reason, I have created a process (in python) which listens to this websocket and sends events to AzureEventHub.
  
What I want to understand is the most appropriate way to deploy the EventHubClient on Azure. Is it to:

- Spin up a VM and run the code there?
- Create a Flask app to host the code?
- Create a Durable Azure Function for that purpose?

**Comment by swathipil** [comment]
@harisbal - We were on vacation/pulled into a few other things. Thanks for your patience on this!
I was taking a look at the SO issue that you linked. I'm not super familiar with Azure Functions, but it looks like they suggested using Azure Functions as well. Would you be able to clarify why Functions wouldn't work in your case?

I'm not sure if you've come across this already, but there is also a list of various app hosting options here: https://docs.microsoft.com/en-ca/azure/developer/python/quickstarts-app-hosting

I'm not sure which would suit your needs best, but please take a look and let us know if you have questions.

Thanks!

**Comment by harisbal** [comment]
Hello @swathipil. No problem, thank you for your attention.
I believe that Azure functions is not appropriate because the websocket connection between the data source and my server needs to be continuous, while Azure Functions are supposed to be completed in short time.
From my understanding, the best solution is to host the service as a webApp but I'm unable to deploy it as a Flask App and I cannot find any relevant documentation.
As a last resort, I have managed to deploy the service in a VM so we can consider the issue closed :)

**Comment by swathipil** [comment]
@harisbal - glad you found a solution! Just in case you're interested, I asked a few other people who are more familiar with hosting options. Here are the main suggestions:

1) Azure Containers - as a simple hosting option: https://docs.microsoft.com/en-us/azure/containers/
2) Using Functions with the WebPubSub extension, which should give you a more persistent connection: https://docs.microsoft.com/en-us/azure/azure-web-pubsub/reference-functions-bindings?tabs=csharp

Thanks!
